spring-cloud-netflix-core/src/main/resources/META-INF/spring.factories

```diff
@@ -15,7 +15,8 @@ org.springframework.cloud.bootstrap.BootstrapConfiguration=\
 org.springframework.cloud.netflix.config.DiscoveryClientConfigServiceBootstrapConfiguration
 
 org.springframework.cloud.client.discovery.EnableDiscoveryClient=\
-org.springframework.cloud.netflix.eureka.EurekaDiscoveryClientConfiguration
+org.springframework.cloud.netflix.eureka.EurekaDiscoveryClientConfiguration,\
+org.springframework.cloud.netflix.eureka.EurekaApplicationEventAutoConfiguration
```

Comment:
You've sold me on moving the `EurekaApplicationEventAutoConfiguration`.  I've made it conditional on `EurekaClient` and `@AutoConfigureAfter(EurekaClientAutoConfiguration.class)`.